custom_components/tesla_custom/teslamate.py

```diff
@@ -194,6 +213,12 @@ async def async_handle_new_data(self, car: TeslaCar, msg: ReceiveMessage):
             self.update_climate_state(car, attr, cast(msg.payload))
             coordinator.async_update_listeners()
 
+        elif mqtt_attr in MAP_CHARGE_STATE:
```

Comment:
Are there any frequent MQTT changes that we may want to explicitly mark as not being saved in the recorder?

Reply:
Not that i've noticed. When teslaMate is connected and working, MQTT updates are about 1/Second
I would think we want the end user to be able to exclude from the recorder, compared to us.

I haven't noticed any extra obseen values in my recorder

Reply:
Sorry, are these entity attributes? I'm referring to the fact that any attribute changes will get logged in the recorder even if the main state doesn't change. This commit shows how to fix it if it does have an attribute that isn't worth recording. https://github.com/home-assistant/core/blob/3e5e937541dd368f3d70c7557ceaa7a07fd284e2/homeassistant/components/unifiprotect/recorder.py

Reply:
So I looked into this.

Unifi is the only non-native component that uses that function (the rest are like Sensor, climate, select, etc...)
So it doesn't really seem like something we should use.

Most of the attributes we're pulling in from TeslaMate end up as States anyway, so Home Assistant's state handling should keep the database in check (aka, don't log if the value hasn't changed)

The only exception to this is
- charger_actual_current
- charger_voltage

And they are only populated While the vechile is charging, which would be getting database updates anyway due to the TeslaCarChargerPower sensor updating

In short, I don't think we need to do the recorder.py for this.

Reply:
Ok, if we're creating state sensors that's fine. The concern is that any time an attribute changes on a sensor, all the data (state and attributes) are recorded in the recorder. If something changes frequently but doesn't really mean anything in the long run, you'd want to exclude that attribute from the recorder. A contrived example might be if you also added a seconds charging attribute to a battery level. Every time the seconds changed, you'd record a new entry even if the battery state may not have changed in that one second.

I thought we were creating a lot more attributes now, so I asked. If that's not the case, then that's fine.

Reply:
So we're good? I don't think there's much else to add to this PR